website/preprints/model.py

```diff
@@ -0,0 +1,189 @@
+import datetime
+import urlparse
+
+from modularodm import fields
+from modularodm.exceptions import ValidationValueError
+
+from framework.celery_tasks.handlers import enqueue_task
+from framework.exceptions import PermissionsError
+from framework.guid.model import GuidStoredObject
+from framework.mongo import ObjectId, StoredObject
+from framework.mongo.utils import unique_on
+from website.files.models import StoredFileNode
+from website.preprints.tasks import on_preprint_updated
+from website.project.model import NodeLog
+from website.project.taxonomies import Subject
+from website.util import api_v2_url
+from website.util.permissions import ADMIN
+from website import settings
+
+@unique_on(['node', 'provider'])
+class PreprintService(GuidStoredObject):
+
+    _id = fields.StringField(primary=True)
+    date_created = fields.DateTimeField(auto_now_add=True)
+    provider = fields.ForeignField('PreprintProvider')
+    node = fields.ForeignField('Node')
+    is_published = fields.BooleanField(default=False)
+    date_published = fields.DateTimeField()
+
+    # This is a list of tuples of Subject id's. MODM doesn't do schema
+    # validation for DictionaryFields, but would unsuccessfully attempt
+    # to validate the schema for a list of lists of ForeignFields.
+    #
+    # Format: [[root_subject._id, ..., child_subject._id], ...]
+    subjects = fields.DictionaryField(list=True)
+
+    @property
+    def preprint_file(self):
```

Comment:
having `preprint_` is duplicative w/ the service class name, any reason to not make it like the other attributes?